p2p/switch.go

```diff
@@ -364,6 +364,13 @@ func (sw *Switch) StopPeerGracefully(peer Peer) {
 }
 
 func (sw *Switch) stopAndRemovePeer(peer Peer, reason interface{}) {
+	if peer == nil {
+		// This condition can occur due to the fact that Switch.OnStop uses stale data and
+		// there is no easy fix for it per https://github.com/cometbft/cometbft/issues/2158
```

Comment:
Should this fix be mentioned in the changelog? Although we are not sure that it indeed solves the problem.

Reply:
No, it doesn't solve the fundamental problem. I agree with you. But it doesn't harm to have this condition imho.